x/distribution/keeper/validator.go

```diff
@@ -86,33 +88,20 @@ func (k Keeper) decrementReferenceCount(ctx sdk.Context, valAddr sdk.ValAddress,
 }
 
 func (k Keeper) updateValidatorSlashFraction(ctx sdk.Context, valAddr sdk.ValAddress, fraction sdk.Dec) {
-	if fraction.GT(sdk.OneDec()) {
-		panic("fraction greater than one")
+	if fraction.GT(sdk.OneDec()) || fraction.LT(sdk.ZeroDec()) {
+		panic(fmt.Sprintf("fraction must be >=0 and <=1, current fraction: %v", fraction))
 	}
-	height := uint64(ctx.BlockHeight())
```

Comment:
Any reason why all this logic was removed? Was calculating `updatedFraction` unnecessary?

Reply:
yeah I think this design decision was overlooked at inception seeing as the revised code eliminates the need for this complexity. By introducing the period into the validator-slash-record-key the fraction no longer needs to be "flattened" with the other fractions in that height, hence we remove all this flattening logic in this function

Reply:
Yes, this is better and simpler. Mea culpa. I think originally we were concerned about iteration costs.